projects/blenderbot2/agents/blenderbot2.py

```diff
@@ -800,7 +800,10 @@ def _set_batch_skip_search(self, valid_exs: List[Message], batch: Batch) -> Batc
 
     def eval_step(self, batch):
         output = super().eval_step(batch)
-        if output is None or not hasattr(self.model, 'retriever'):
+        model = self.model_api
+        if isinstance(model, torch.nn.parallel.DistributedDataParallel):
+            model = model.module
+        if output is None or not hasattr(model, 'retriever'):
```

Comment:
sorry, I might not have been clear enough: `model_api` [already takes care of this check](https://github.com/facebookresearch/ParlAI/blob/66c71e01b79b118a717c1115fecea5e99b12429a/parlai/agents/rag/rag.py#L141-L146)

```suggestion
        if output is None or not hasattr(self.model_api, 'retriever'):
```
This is the only change required

Reply:
ah didn't know that! thanks for the pointer